src/main/java/com/google/sps/servlets/EnterQueue.java

```diff
@@ -61,15 +74,36 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
             Key classKey = KeyFactory.stringToKey(classCode);
             Entity classEntity = datastore.get(txn, classKey);
 
-            ArrayList<String> updatedQueue = (ArrayList) classEntity.getProperty("studentQueue");
-
-            String visitCode = (String) classEntity.getProperty("visitKey");
-            Entity visitEntity = datastore.get(txn, KeyFactory.stringToKey(visitCode));
+            // Get date
+            LocalDate localDate = LocalDate.now(clock);
+            ZoneId defaultZoneId = ZoneId.systemDefault();
+            Date currDate = Date.from(localDate.atStartOfDay(defaultZoneId).toInstant());
+
+            // Query visit entity for particular day
+            Filter classVisitFilter =
+                new FilterPredicate("classKey", FilterOperator.EQUAL, classKey);
+            Filter dateVisitFilter = new FilterPredicate("date", FilterOperator.EQUAL, currDate);
+            CompositeFilter visitFilter =
+                CompositeFilterOperator.and(dateVisitFilter, classVisitFilter);
+            Query query = new Query("Visit").setFilter(visitFilter);
+
+            // Get visit entity for particular day
+            Entity visitEntity;
+            if (datastore.prepare(query).countEntities() == 0) {
```

Comment:
It would be more efficient to prepare the query only once and then run `counEntities` and `asSingleEntity` on the same `PreparedQuery`